### PR TITLE
main: Make version bits GUI warning clearer to translators

### PR DESCRIPTION
**Pull request by laanwj**
Trivial message changes to make translation easier 
- main: Make version bits GUI warning more clear to translators (reported by yahoe.001 on Transifex for the French translation)
- init: Fix typo in help message for -whitelistforcerelay (reported by pryds on Transifex in the Danish translation)

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -2596,13 +2596,13 @@ void static UpdateTip(CBlockIndex *pindexNew, const CChainParams& chainParams) {
             ThresholdState state = checker.GetStateFor(pindex, chainParams.GetConsensus(), warningcache[bit]);
             if (state == THRESHOLD_ACTIVE || state == THRESHOLD_LOCKED_IN) {
                 if (state == THRESHOLD_ACTIVE) {
-                    strMiscWarning = strprintf(_("Warning: unknown new rules activated (versionbit %i)"), bit);
+                    strMiscWarning = strprintf(_("Warning: unknown new rules activated (version bit %i)"), bit);
```
Comment (paveljanik) [inline src/main.cpp]: `p2p-versionbits-warning.py` has to be adapter to this change 8)
```python
self.vb_pattern = re.compile("^Warning.*versionbit")
```

**Comment by paveljanik** [comment]
Concept ACK.

**Comment by jonasschnelli** [comment]
utACK ac0730c16dc6eeb5fcb88aa136ef4cc56e360414

**Comment by laanwj** [comment]
Why are we translating this at all if we match on the English text?

This is the wrong thing to do. For example, running this test against bitcoin-qt with non-English locale would also break this test.
(or in the hypothetical case we ever implement #6092)

**Comment by laanwj** [comment]
I remember solving a similar issue in #7141, and the outcome was:

- The warning shown in the GUI can be translated (but only if it is a common warning, for google-ability)
- The warning shown in getinfo should not be translated - as it could be matched by regexps and such

**Comment by paveljanik** [comment]
This definitely is not nice, yes.

**Comment by sipa** [comment]
Anyone mind having a look at the Travis failure?

**Comment by paveljanik** [comment]
The failures are because of https://github.com/bitcoin/bitcoin/pull/7995#discussion_r61924699

**Comment by sipa** [comment]
@paveljanik Thanks.

@laanwj Feel like addressing that?

**Comment by laanwj** [comment]
I don't want to change the test as this is not supposed to be an API change. All I intended to do was change the GUI (translation) message.

I don't, however, think this message should be translated at all. Especially as it is reported on RPC.

**Comment by laanwj** [comment]
Removed version bits warnings as translation message, this should pass now.

File: src/main.cpp [hunk 2]
```diff
@@ -2596,7 +2596,7 @@ void static UpdateTip(CBlockIndex *pindexNew, const CChainParams& chainParams) {
             ThresholdState state = checker.GetStateFor(pindex, chainParams.GetConsensus(), warningcache[bit]);
             if (state == THRESHOLD_ACTIVE || state == THRESHOLD_LOCKED_IN) {
                 if (state == THRESHOLD_ACTIVE) {
-                    strMiscWarning = strprintf(_("Warning: unknown new rules activated (versionbit %i)"), bit);
+                    strMiscWarning = strprintf("Warning: unknown new rules activated (versionbit %i)", bit);
```
Comment (paveljanik) [inline src/main.cpp]: Removing translation here results in this message being shown untranslated in the UI - try it now on testnet.

Comment (sipa) [inline src/main.cpp]: @laanwj Can you have a look?

Comment (MarcoFalke) [inline src/main.cpp]: If this is the intent, please change the pull subject.

Comment (laanwj) [inline src/main.cpp]: Well the point is that it should not appear translated on the API (RPC).
There is currently no way to set a different message here for the GUI and for the RPC.
There should be one, but the point here was to do a quick change. I don't think it's worth spending a lot of time here now, so I'm going to close this.

**Comment by laanwj** [comment]
Going to close this, I don't care enough to .

**Pull request closed without merging**